src/printer.js

```diff
@@ -152,24 +130,138 @@ function isLastStatement(path) {
   return body[body.length - 1] === node;
 }
 
-function printLines(path, options, print) {
-  const text = options.originalText;
-  const printed = path.map(stmtPath => {
-    const stmt = stmtPath.getValue();
-    const parts = [];
-    parts.push(print(stmtPath));
-    if (lineShouldEndWithSemicolon(stmtPath)) {
-      parts.push(";");
-    }
-    if (!isLastStatement(stmtPath)) {
-      if (sharedUtil.isNextLineEmpty(text, stmt, options)) {
-        parts.push(hardline);
-      }
-      parts.push(hardline);
+function nodeHasStatement(node) {
+  return [
+    "block",
+    "program",
+    "namespace",
+    "class",
+    "interface",
+    "trait",
+    "traituse",
+    "declare"
+  ].includes(node.kind);
+}
+
+function isFirstNodeInParentProgramNode(path) {
```

Comment:
would all of these functions make sense to go in the util file? the printer is getting crazy long

Reply:
:+1: